tests/test_config.py

```diff
@@ -0,0 +1,29 @@
+import sys
+from pathlib import Path
+
+import pytest
+from pytest import LogCaptureFixture, MonkeyPatch
+
+from lakefs_spec.config import LakectlConfig
+
+
+def test_config_read_nonexistent_file():
+    """
+    Tests that a FileNotFoundError is raised when a lakectl config is read from
+    a nonexistent file.
+    """
+    with pytest.raises(FileNotFoundError):
+        LakectlConfig.read("aaaaaaaaaaaaaaaa.yaml")
+
+
+def test_lakectl_config_parsing_without_yaml(
+    monkeypatch: MonkeyPatch, caplog: LogCaptureFixture, temporary_lakectl_config: str
+) -> None:
+    # unset YAML module from sys.modules.
+    monkeypatch.setitem(sys.modules, "yaml", None)
+
+    LakectlConfig.read(Path("~/.lakectl.yaml").expanduser())
```

Comment:
Does that not implicitly assume that the config file must exist on the machine to not spuriously fail the test?

Reply:
Setting up the config file (temporarily) is the exact point of the `temporary_lakectl_config` fixture, see `conftest.py`. So yes, it does assume, but rightfully so, since it is created in the fixture setup.